File: frontend-web/src/types/project.ts
export type ProjectStatus = 'PLANNING' | 'IN_PROGRESS' | 'REVIEW' | 'DONE';
export type TaskStatus = 'TODO' | 'DOING' | 'DONE';

export interface UserInfo {
    id: string;
    name: string;
    email: string;
    avatar?: string; // Optional for frontend display
    role?: string;
}

export interface Project {
    id: string;
    title: string;
    client: string;
    budget: number;
    startDate: string;
    endDate: string;
    status: ProjectStatus;
    progress: number;
    description?: string;
    team: UserInfo[];
    createdAt?: string;
    updatedAt?: string;
}

export interface Task {
    id: string;
    projectId: string;
    title: string;
    status: TaskStatus;
    assignee?: UserInfo;
    orderIndex?: number;
    createdAt?: string;
    updatedAt?: string;
}

File: backend/src/main/java/com/erp/entity/Project.java
package com.erp.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.HashSet;
import java.util.Set;

/**
 * @file Project.java
 * @description 專案實體 / Project Entity
 * @description_en Represents a project with details, status, and team members
 * @description_zh 專案基本資料與團隊成員關聯
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "projects")
@EntityListeners(AuditingEntityListener.class)
public class Project {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private String id;

    @Column(nullable = false)
    private String title;

    @Column(nullable = false)
    private String client;

    @Column(nullable = false)
    private Double budget;

    @Column(nullable = false)
    private LocalDate startDate;

    @Column(nullable = false)
    private LocalDate endDate;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private ProjectStatus status;

    @Column(nullable = false)
    private Integer progress;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column
    private String githubRepo;

    @Column
    private String githubBranch;

    @Column(columnDefinition = "TEXT")
    private String backupConfig;

    @Column
    private String githubToken;

    @Column
    private String fileLocation;

    @ManyToMany(fetch = FetchType.EAGER)
    @JoinTable(name = "project_members", joinColumns = @JoinColumn(name = "project_id"), inverseJoinColumns = @JoinColumn(name = "user_id"))
    @Builder.Default
    private Set<User> team = new HashSet<>();

    @CreatedDate
    @Column(nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @LastModifiedDate
    @Column(nullable = false)
    private LocalDateTime updatedAt;
}
